src/main/java/com/squareup/javapoet/CodeBlock.java

```diff
@@ -269,6 +278,13 @@ public Builder addStatement(String format, Object... args) {
       return this;
     }
 
+    public Builder addStatement(CodeBlock codeBlock) {
```

Comment:
I have noticed that now that I don't use that method as I've replaced it with "format with CodeBlock s" but let assume we generate code with the following func (as usual smth you don't want to write by yourself)
```java
        public static CodeBlock composeObjectGetter(CodeBlock leftHandSide, CodeBlock identifier, String... properties) {
        CodeBlock.Builder builder = leftHandSide.toBuilder();
        for (int i = properties.length - 1; i >= 0; i--) {
            builder.add("(($T) ", ClassName.get(JSObject.class));
        }
        builder.add(identifier);
        for (int i = 0; i < properties.length; i++) {
            builder.add(OBJECT_GETTER, properties[i]);
        }
        return builder.build();
    }
```

then you might want to add it as Statement but I haven't found a method to do that. Is there any other way? btw at first I used String for generation and then just added it but unfortunately I didn't get any class imports with that and haven't found a way to import manually.

Now I generate only right hand side and the use something like `builder.addStatement("$L = ($B).something()", name, codeblock)`